### PR TITLE
Fix screenshot example

### PR DESCRIPTION
**Pull request by akimakinai**
# Objective

I noticed some issues in `screenshot` example:
1. Cursor icon won't return from `SystemCursorIcon::Progress` to default icon, even though screen shot saving is done.
2. Panics when exiting window: ``called `Result::unwrap()` on an `Err` value: NoEntities("bevy_ecs::query::state::QueryState<bevy_ecs::entity::Entity, bevy_ecs::query::filter::With<bevy_window::window::Window>>")``

## Solution

1. Caused by cursor updating system not responding to [`CursorIcon` component removal](https://github.com/bevyengine/bevy/blob/5cfcbf47ed386515adefe80f99f8e8f7e7f3ce5a/examples/window/screenshot.rs#L38). I believe it should, so change it to react to `RemovedComponents<CursorIcon>`. (a suggestion)
2. Use `get_single` for window.

## Testing

- run screenshot example

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Good fixes, thank you!

File: crates/bevy_render/src/view/window/cursor.rs [hunk 1]
```diff
@@ -73,12 +74,22 @@ pub enum CustomCursor {
 
 pub fn update_cursors(
     mut commands: Commands,
-    mut windows: Query<(Entity, Ref<CursorIcon>), With<Window>>,
+    windows: Query<(Entity, Ref<CursorIcon>), With<Window>>,
+    mut icon_removed: RemovedComponents<CursorIcon>,
```
Comment (janhohenheim) [inline crates/bevy_render/src/view/window/cursor.rs:78]: 'RemovedComponents` is on the chopping block. Could you replace this with an `OnRemove` observer?

**Review by janhohenheim — CHANGES_REQUESTED** [review]
Looks nearly ready, thanks! I'd appreciate if you could use an observer here, so we don't have to fix this usage of `RemovedComponents` when we deprecate it :)

**Review by eero-lehtinen — APPROVED** [review]

**Review by janhohenheim — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@akimakinai it looks like I didn't resolve merge conflicts fully correctly in the web UI: there's an unused import. When you get a chance can you fix this locally and push the changes? Once that's done I'll merge this in.

**Pull request merged**